libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -389,9 +415,94 @@ def test_that_same_name_vector_store_can_be_created_in_different_schemas() -> No
+    # Create a SQLServer_VectorStore without `embedding_length` defined.
+    store.add_texts(texts)
+
+    # Add texts using an embedding function with a different length.
+    # This should raise an exception.
+    #
+    store.embedding_function = FakeEmbeddings(size=420)
+    with pytest.raises(Exception):
+        store.add_texts(texts)
+
+
+def test_that_similarity_search_returns_expected_no_of_documents(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Test that the amount of documents returned when similarity search
+    is called is the same as the number of documents requested."""
+    store.add_texts(texts)
+    number_of_docs_to_return = 3
+    result = store.similarity_search(query="Good review", k=number_of_docs_to_return)
+    assert len(result) == number_of_docs_to_return
+
+
+def test_that_similarity_search_returns_results_with_scores_sorted_in_ascending_order(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Assert that the list returned by a similarity search
+    is sorted in an ascending order. The implication is that
+    we have the smallest score (most similar doc.) returned first.
+    """
+    store.add_texts(texts)
+    number_of_docs_to_return = 4
+    doc_with_score = store.similarity_search_with_score(
+        "Good review", k=number_of_docs_to_return
+    )
+    assert doc_with_score == sorted(doc_with_score, key=lambda x: x[1])
+
+
+def test_that_case_sensitivity_does_not_affect_distance_strategy(
+    texts: List[str],
+) -> None:
+    """Test that when distance strategy is set on a case sensitive DB,
+    a call to similarity search does not fail."""
+    connection_string_to_master = "mssql+pyodbc://@localhost/master?driver=ODBC+Driver+17+for+SQL+Server&Trusted_connection=yes"
+
+    conn = create_engine(connection_string_to_master).connect()
+    conn.rollback()
+
+    if conn.connection.dbapi_connection is not None:
+        conn.connection.dbapi_connection.autocommit = True
+
+    conn.execute(text(_CREATE_COLLATION_DB_QUERY))
+    conn.execute(text(f"use {_COLLATION_DB_NAME}"))
+
+    store = SQLServer_VectorStore(
+        connection=conn,
+        connection_string=connection_string_to_master,
+        # FakeEmbeddings returns embeddings of the same size as `embedding_length`.
+        embedding_length=EMBEDDING_LENGTH,
+        embedding_function=FakeEmbeddings(size=EMBEDDING_LENGTH),
+        table_name=_TABLE_NAME,
+    )
+    collation_query_result = (
+        conn.execute(text(_COLLATION_QUERY % (_COLLATION_DB_NAME))).fetchone()
+    )  # Sample return value: ('LangChainVectors', 'SQL_Latin1_General_CP1_CS_AS')
+
+    assert (
+        collation_query_result is not None
+    ), "No collation data returned from the database."
+    # Confirm DB is case sensitive
+    assert "_CS" in collation_query_result.collation_name
+
+    store.add_texts(texts)
+    store._distance_strategy = DistanceStrategy.DOT
```

Comment:
another test would be to explicitly set the _distance_strategy to UPPERCASE and ensure the result is the same as that you would get with the lower case distance strategy

Reply:
I tried that and was flagged by the linting tool. This is because `_distance_strategy` expects a type `DistanceStrategy` and doing the conversion returns a `string`.